src/server/connection.rs

```diff
@@ -3126,6 +3164,54 @@ impl Connection {
         self.inner.send(msg.into());
         self.supported_encoding_flag = (true, not_use);
     }
+
+    #[cfg(not(any(target_os = "android", target_os = "ios")))]
+    async fn handle_window_focus(&mut self) {
+        if let Some(s) = self.server.upgrade() {
+            let server = s.write().unwrap();
+            let subbed_disp_count = server.get_subbed_displays_count(self.inner.id());
+            if subbed_disp_count != 1 {
+                return;
+            }
+        }
+        let displays = super::display_service::get_sync_displays();
+        let current_display = crate::get_focused_display(displays);
+        if let Some(d_index) = current_display {
+            let mut misc = Misc::new();
+            misc.set_follow_current_display(d_index as i32);
+            let mut msg_out = Message::new();
+            msg_out.set_misc(misc);
+            self.send(msg_out).await;
+        }
+    }
+
+    #[cfg(not(any(target_os = "android", target_os = "ios")))]
+    async fn handle_cursor_switch_display(&mut self, pos: CursorPosition) {
+        if let Some(s) = self.server.upgrade() {
+            let server = s.write().unwrap();
+            let subbed_disp_count = server.get_subbed_displays_count(self.inner.id());
```

Comment:
No way to mark this status on Connection structure? Checking it every time is not good.

Reply:
New changes:
- Remove windowfocus protobuf message
- New bool field `multi_ui_session`. True when `get_subbed_displays_count > 1`
- Unsub window focus if `multi_ui_session`